product_packaging_type/models/product_packaging_type.py

```diff
@@ -51,7 +51,8 @@ def default_packaging_type_id(self):
         ondelete="restrict",
         default=lambda p: p.default_packaging_type_id(),
     )
-    type_has_gtin = fields.Boolean(related="packaging_type_id.has_gtin", readonly=True)
+    type_has_gtin = fields.Boolean(readonly=True, compute="_compute_type_has_gtin")
```

Comment:
Semantically, this is very weird that this field is named `type_has_gtin`, as it *may* have a gtin according to the implementation of `_compute_type_has_gtin`. This is the kind of thing that may hurt later when someone assumes that `type_has_gtin` really has a gtin for another part of logic. Since the only goal of this field is to set the field barcode conditionally, I would name it `barcode_required_for_gtin`.